module/zfs/dsl_dataset.c

```diff
@@ -2429,7 +2429,8 @@ get_receive_resume_stats_impl(dsl_dataset_t *ds)
 
 		str = kmem_alloc(compressed_size * 2 + 1, KM_SLEEP);
 		for (int i = 0; i < compressed_size; i++) {
-			(void) sprintf(str + i * 2, "%02x", compressed[i]);
+			(void) snprintf(str + i * 2, compressed_size * 2 + 1,
```

Comment:
Another possible candidate for strlcat; failing that, the size should be (compressed_size * 2 + 1) - i*2, or something like that.